libraries/AP_Scripting/AP_Scripting_MAVLink.cpp

```diff
@@ -0,0 +1,181 @@
+/*
+   This program is free software: you can redistribute it and/or modify
+   it under the terms of the GNU General Public License as published by
+   the Free Software Foundation, either version 3 of the License, or
+   (at your option) any later version.
+
+   This program is distributed in the hope that it will be useful,
+   but WITHOUT ANY WARRANTY; without even the implied warranty of
+   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+   GNU General Public License for more details.
+
+   You should have received a copy of the GNU General Public License
+   along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+/*
+  Scripting MAVLink class, to generate MAVLink packets and decode incoming
+  MAvLink data. It is designed for High Latency MAVLink connections, in most cases
+  users should use a regular MAVLink serial port.
+ */
+#include "AP_Scripting_MAVLink.h"
+
+#if HAL_HIGH_LATENCY2_ENABLED
+ScriptingMAVLink::ScriptingMAVLink(bool use_mavlink1)
+{
+    // Instantiate a ScriptingMAVLink and set the MAVLink version
+    // true to use MAVLink1, false to use MAVLink2
+
+    useMAVLink1 = use_mavlink1;
+
+#if CONFIG_HAL_BOARD == HAL_BOARD_SITL
+    if (_singleton != nullptr) {
+        AP_HAL::panic("ScriptingMAVLink must be singleton");
+    }
+#endif
+    _singleton = this;
+};
+
+bool ScriptingMAVLink::is_high_latency_enabled()
+{
+    // Return true if high latency mode is enabled
+    GCS_MAVLINK *link = gcs().chan(0);
+    return link->high_latency_link_enabled;
+}
+
+void ScriptingMAVLink::set_high_latency_enabled(bool hl_enabled)
+{
+    //enable or disable high latency mode
+    GCS_MAVLINK *link = gcs().chan(0);
+    mavlink_command_long_t pkt;
+    pkt.param1 = hl_enabled;
+
+    link->handle_control_high_latency(pkt);
+}
+
+uint16_t ScriptingMAVLink::create_high_latency2(uint8_t* msgbuf)
```

Comment:
I think this should be done in the script. The ultimate solution would be to directly generate the lua as we do for python and c++.

Reply:
https://github.com/ArduPilot/pymavlink/pull/386